semgrep/semgrep/commands/app_scan.py

```diff
@@ -0,0 +1,122 @@
+import sys
+import time
+from pathlib import Path
+from typing import Any
+from typing import Iterable
+from typing import Sequence
+
+import click
+
+from semgrep.commands.login import Authentication
+from semgrep.commands.scan import CONTEXT_SETTINGS
+from semgrep.commands.scan import scan
+from semgrep.commands.scan import scan_options
+from semgrep.error import FATAL_EXIT_CODE
+from semgrep.error import SemgrepError
+from semgrep.ignores import IGNORE_FILE_NAME
+from semgrep.semgrep_app import ScanHandler
+
+# These patterns are excluded via --exclude regardless of other ignore configuration
+ALWAYS_EXCLUDE_PATTERNS = [".semgrep/", ".semgrep_logs/"]
+
+# These patterns are excluded via --exclude unless the user provides their own .semgrepignore
+DEFAULT_EXCLUDE_PATTERNS = ["test/", "tests/", "*_test.go"]
+
+
+def yield_valid_patterns(patterns: Iterable[str]) -> Iterable[str]:
+    for pattern in patterns:
+        pattern = pattern.strip()
+
+        if not pattern:
+            continue
+        if pattern.startswith("#"):
+            continue
+
+        yield pattern
+
+
+def yield_exclude_args(requested_patterns: Sequence[str]) -> Iterable[str]:
+    patterns = [*yield_valid_patterns(requested_patterns), *ALWAYS_EXCLUDE_PATTERNS]
+    if Path(IGNORE_FILE_NAME).is_file():
+        patterns.extend(DEFAULT_EXCLUDE_PATTERNS)
+
+    for pattern in patterns:
+        yield from ["--exclude", pattern]
```

Comment:
```suggestion
    yield from patterns
```

even better if this is updated with a new name and to return a Sequence.